### ISSUE TITLE
Can create illegally-named blob containers

### ISSUE THREAD

**Issue by jumpinjackie** [Azure/Azurite#456]
### Which service(blob, file, queue, table) does this issue concern?

blob

### Which version of the Azurite was used?

Whatever version/tag `mcr.microsoft.com/azure-storage/azurite:latest` is currently (as of 15th June 2020)

### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)

I'm pulling the azurite docker image

### What's the Node.js version?

Whatever came with this docker image

### What problem was encountered?

When creating a new blob container against the azurite emulator, it does not enforce the [container naming restrictions](https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata)

In my case, I successfully made a container named `dev_public`, which should have failed because the name contained an underscore which is not allowed per the above naming restrictions.

This illegally-named container proceed to blow up container listing operations. For example, expanding the emulator storage accounts on Microsoft Azure Storage Explorer will error with

```
Unable to retrieve child resources.

Details:
Container name format is incorrect.
```

### Steps to reproduce the issue?

1. Create a container with one or more illegal characters in the name (eg. An underscore)
    Expect: The container creation API to fail on this name
    Get: The container creation API succeeds on this name
2. Open MS Azure Storage Explorer and expand the **Blob Containers** section of the emulator storage account. The listing operation fails with the above message due to the illegally-named container that azurite let through.

### Have you found a mitigation/solution?

I had to trash my running azurite container and restart a new one with a personal mental note to [respect the naming restrictions on blob containers](https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata) because azurite failed to emulate this aspect properly.

**Comment by blueww** [comment]
@jumpinjackie
Thanks for reporting this issue!

We can repro it and will priority the fix of it in a future release.

**Comment by abhi2495** [comment]
Yes same issue. There seems to be no validation on naming of containers while creating them. I am using Azurite node in my Java integration tests

**Comment by stingster** [comment]
Any update on this issue?

**Comment by blueww** [comment]
@stingster 

Thanks for the checking!
We are busy on some other stuffs, so still have not start this item.

We welcome contribute to Azurite. 
If you would like to contribute, you can add the container name check in context middleware : https://github.com/xiaoningliu/Azurite/blob/4ef9204082a21c3ad5f72c7800940c3282ffb1d6/src/blob/middlewares/blobStorageContext.middleware.ts#L66

**Comment by arian2ashk** [comment]
The issue also exist for Table service

**Comment by blueww** [comment]
The issue has been fixed in Azurite 3.13.0
